.github/workflows/gradle-build-scan.yml

```diff
@@ -0,0 +1,98 @@
+name: Upload Gradle build scans to ge.armeria.dev
+
+on:
+  workflow_run:
+    workflows: [CI]
+    types:
+      - completed
+
+env:
+  LC_ALL: "en_US.UTF-8"
+  GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
+  RUN_ID: ${{ github.event.workflow_run.id }}
+  COMMIT_SHA: ${{ github.event.workflow_run.head_sha }}
+  # Used by Octokit
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  upload-gradle-build-scan:
+    name: Upload Gradle build scans
+    if: github.repository == 'line/armeria'
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
+
+      - id: setup-jdk-19
+        name: Set up JDK 19
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '19'
+
+      - name: Download artifact
+        id: download-artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          workflow_conclusion: ""
+          run_id: ${{ env.RUN_ID }}
+          name: build-scan.*
+          name_is_regexp: true
+          path: build-scans
+          check_artifacts: true
+          search_artifacts: true
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+
+      - id: get-pr-number-by-sha
+        name: Get PR number by SHA
+        working-directory: .github/actions
+        run: |
+          export NVM_DIR=~/.nvm
+          source ~/.nvm/nvm.sh
+          npm install
+          npm run get-pr-number-by-sha
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+
+      - id: upload-build-scans
+        name: Upload build scans
+        run: |
+          DOWNLOAD_DIR="build-scans/"
+          BUILD_SCAN_DIR="${HOME}/.gradle/build-scan-data"
```

Comment:
Is there a way to upload the build scans at a specified directory other than the default location? We could upload from `build-scans/` without copying them to a shared directory (`~/.gradle/build-scan-data`), which could potentially cause a race condition.

Reply:
When I read the Javadoc of `buildScan`, there is no option to customize the folder `(~/.gradle/build-scan-data)`.
As GitHub Actions executes only one run in an instance, we would not encounter race conditions at the moment. That said, it sounds very helpful if we can specify the folder.

@ribafish Please correct me if I'm wrong.

Reply:
Correct, there's no way to specifiy a different directory to upload the build scans from as such. If you wanted to run these in parallel, you could create copies of the project to run parallel builds while specifying a different gradle user home directory for each build with `-Dgradle.user.home=(path to directory).`